examples/schema_site_vrf_region/main.tf

```diff
@@ -13,16 +13,48 @@ provider "mso" {
   insecure = true
 }
 
-resource "mso_schema_site_vrf_region" "vrfRegion" {
-  schema_id          = "5efd6ea60f00005b0ebbd643"
-  template_name      = "Template1"
-  site_id            = "5efeb3c4190000cc12d05376"
-  vrf_name           = "Myvrf"
+data "mso_site" "demo_site" {
+  name = "demo_site"
+}
+
+resource "mso_tenant" "demo_tenant" {
+  name = "demo_tenant"
+}
+
+resource "mso_schema" "demo_schema" {
+  name = "demo_schema"
+  template {
+    name         = "Template1"
+    display_name = "Template1"
+    tenant_id    = mso_tenant.demo_tenant.id
+  }
+}
+
+resource "mso_schema_site" "demo_schema_site" {
+  schema_id     = mso_schema.demo_schema.id
+  template_name = one(mso_schema.demo_schema.template).name
+  site_id       = data.mso_site.demo_site.id
+}
+
+resource "mso_schema_template_vrf" "vrf1" {
+  schema_id              = mso_schema.demo_schema.id
+  template               = one(mso_schema.demo_schema.template).name
+  name                   = "vrf1"
+  display_name           = "vrf1"
+  ip_data_plane_learning = "enabled"
+}
+
+// AZURE specific example
+resource "mso_schema_site_vrf_region" "vrf_region_azure" {
+  schema_id          = mso_schema.demo_schema.id
+  template_name      = one(mso_schema.demo_schema.template).name
+  site_id            = data.mso_site.demo_site.id
```

Comment:
Would it be better to reference site_id here with mso_schema_site.demo_schema_site.id rather than data.mso_site.demo_site.id?

Reply:
ok